pkg/controller/controller.go

```diff
@@ -397,15 +417,30 @@ func (s *Service) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest
 	var volumeResponse *csi.Volume
 	resp, err := creator.Create(ctx, req, sizeInBytes, arr.GetClient())
 	if err != nil {
-		if apiError, ok := err.(gopowerstore.APIError); ok && (apiError.VolumeNameIsAlreadyUse() || apiError.FSNameIsAlreadyUse()) {
-			volumeResponse, err = creator.CheckIfAlreadyExists(ctx, req.GetName(), sizeInBytes, arr.GetClient())
-			if err != nil {
-				return nil, err
+		if apiError, ok := err.(gopowerstore.APIError); ok {
+			switch {
+			case apiError.FSCreationLimitReached() && useNFS:
+				arr.NASCooldownTracker.MarkFailure(selectedNasName)
+				return nil, status.Error(codes.ResourceExhausted, err.Error())
+
+			case apiError.VolumeNameIsAlreadyUse(), apiError.FSNameIsAlreadyUse():
+				volumeResponse, err = creator.CheckIfAlreadyExists(ctx, req.GetName(), sizeInBytes, arr.GetClient())
+				if err != nil {
+					return nil, err
+				}
+
+			default:
+				// Catch all other API errors
+				return nil, status.Error(codes.Internal, err.Error())
```

Comment:
After giving it more thought, I realized that we should treat a NAS as failed if the FS creation doesn't succeed, rather than only considering it a failure when the FS limit is reached. 
Please update the UT to reflect this change. 
```suggestion
		if useNFS {
			arr.NASCooldownTracker.MarkFailure(selectedNasName)
			return nil, status.Error(codes.ResourceExhausted, err.Error())
		}
		if apiError, ok := err.(gopowerstore.APIError); ok && (apiError.VolumeNameIsAlreadyUse() || apiError.FSNameIsAlreadyUse()) {
			volumeResponse, err = creator.CheckIfAlreadyExists(ctx, req.GetName(), sizeInBytes, arr.GetClient())
			if err != nil {
				return nil, err
```